src/org/thoughtcrime/securesms/database/ThreadDatabase.java

```diff
@@ -528,6 +558,13 @@ public long getThreadIdFor(Recipient recipient, int distributionType) {
   }
 
   public @Nullable Recipient getRecipientForThreadId(long threadId) {
+    // Loki - Cache the address.
+    // Don't know if this will affect any other signal code
+    // Don't know if it is necessary to add some cache time
+    if (addressCache.containsKey(threadId) && addressCache.get(threadId) != null) {
+      return Recipient.from(context, addressCache.get(threadId), false);
```

Comment:
we may want to remove address from cache on `deleteConversation` and `deleteConversations`